### PR TITLE
boards: convert images to JPEG and reduce image size

### LINKED ISSUE #42356: Repo size - board documentation - large PNGs
A ZIP of the current repo is about 80 MB. This is painful for everyone new to the project, especially those not fortunate enough to have a decent Internet connection, but further for every code drop, fork, etc. Even worse it won't ever go away (files are kept in history and will be checked out, even if removed).

This is mostly attributable to images in the 'documentation' for the boards. Sometimes entirely pointless images, like boards/arm/blackpill_f411ce/doc/img/blackpill-v2.jpg, and this practice is questionable in general: Why can't we expect developers or users to look this up from the original sources, i.e. the manufacturer? (Avoiding the whole 'storing PNGs in a git-repo is rather pointless'-discussion for now).
It is rather wasteful to need to push around and store 64 MBs of images that will most certainly almost never be used to write code or deploy to devices.

Some of these images easily approach 1 MB, most have hundreds of KB for very simple content, and considering that there are and will be quite a lot of boards, the repository will certainly grow further.

I think it is too late to remove all the offending images (rewriting history of such a popular repository would be rather painful I fear), I think there should be at least safeguards implemented to disallow merging more image files, or if really necessary, at least only if they are below a certain size (50kB?) -- preferably they should be vector drawings anyway?

I couldn't find anything in the guidelines, so please consider adding this wherever appropriate and sensitize the maintainers for this issue.

### PR DESCRIPTION
**Pull request by gmarull**
The boards folder uses ~142.8 MB, being the largest in the repository.
This is due mostly to board images, which are in most cases not
optimized for web content. This patch tries to address this problem by
converting all pictures to JPEG (quality 75) and by adjusting its size
up to 750 px (the width of the documentation content). Images that
specified a fixed width in rst files are converted down to that value
instead.

With this patch, folder goes down to ~52.6 MB from 142.8 MB (-~63%).
Note that this patch introduces a new set of binary files to git
history, though (bad).

The process has been automated using a Python script (see commit for details)

This also makes me wonder if docs should live in its own repo as many projects do.

**PLEASE REVIEW RESULTS HERE:**
https://builds.zephyrproject.io/zephyr/pr/49024/docs/

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/42356

### CODE REVIEW

**Comment by galak** [comment]
We really need a CI check for this @stephanosio

**Comment by stephanosio** [comment]
> We really need a CI check for this @stephanosio - the file size of image files.

Please file an issue for this if you think this is necessary.

**Comment by galak** [comment]
> Please file an issue for this if you think this is necessary.

which repo are we using for such issues?

**Comment by carlescufi** [comment]
> > Please file an issue for this if you think this is necessary.
> 
> which repo are we using for such issues?

Main zephyr repo @galak

**Comment by galak** [comment]
> > > Please file an issue for this if you think this is necessary.
> > 
> > 
> > which repo are we using for such issues?
> 
> Main zephyr repo @galak

https://github.com/zephyrproject-rtos/zephyr/issues/49026

File: boards/posix/native_posix/doc/index.rst [hunk 1]
```diff
@@ -351,7 +351,7 @@ side-effects.
 Architecture
 ************
 
-.. figure:: native_layers.svg
+.. figure:: native_layers.jpg
```
Comment (aescolar) [inline boards/posix/native_posix/doc/index.rst:354]: You don't want to convert vector graphic files. (they tend to be much smaller, they look better, they are editable and may be the source itself for the drawing)

**Review by aescolar — DISMISSED** [review]

**Review by evgeniy-paltsev — DISMISSED** [review]
As I can see some of the images were significantly reduced in size, so now it's really hard to read some captions on the picture. I.e: https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/eccc3235c8bdcf378dbf8c0a41db578c5222614a/boards/arc/hsdk/doc/mikrobus_header.jpg

**Comment by gmarull** [comment]
> As I can see some of the images were significantly reduced in size, so now it's really hard to read some captions on the picture. I.e: [`eccc323`/boards/arc/hsdk/doc/mikrobus_header.jpg (raw)](https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/eccc3235c8bdcf378dbf8c0a41db578c5222614a/boards/arc/hsdk/doc/mikrobus_header.jpg)

The script resizes images to the specified width in rst, and if you check the docs, you'll see it actually looks the same:

image
https://docs.zephyrproject.org/latest/boards/arc/hsdk/doc/index.html

So, in this case, it looks like the width in the rst needs to be fixed.

File: boards/posix/native_posix/doc/index.rst (hunk 1 above)
Comment (gmarull) [reply boards/posix/native_posix/doc/index.rst:354]: will fix that, it's an issue in the script thanks for spotting this one

**Review by firscity — CHANGES_REQUESTED** [review]
Xen Project picture in xenvm docs probably was converted to JPG incorrectly and now is distorted. Please check this, thanks.
https://builds.zephyrproject.io/zephyr/pr/49024/docs/boards/arm64/xenvm/doc/index.html?highlight=xenvm

**Comment by gmarull** [comment]
> Xen Project picture in xenvm docs probably was converted to JPG incorrectly and now is distorted. Please check this, thanks. [builds.zephyrproject.io/zephyr/pr/49024/docs/boards/arm64/xenvm/doc/index.html?highlight=xenvm](https://builds.zephyrproject.io/zephyr/pr/49024/docs/boards/arm64/xenvm/doc/index.html?highlight=xenvm)

that's an interesting artifact. Btw, do we really need logos in board docs? Does this add any value?

**Comment by firscity** [comment]
> that's an interesting artifact. Btw, do we really need logos in board docs? Does this add any value?

If repo size should be decreased - my opinion is no, we don't. At least we definitely don't want to have distorted images in docs.

**Review by firscity — DISMISSED** [review]
Xenvm part LGTM

File: boards/x86/qemu_x86/doc/index.rst [hunk 2]
```diff
@@ -8,13 +8,6 @@ Overview
 
 The X86 QEMU board configuration is used to emulate the X86 architecture.
 
-.. figure:: qemu_x86.png
-   :width: 600px
-   :align: center
-   :alt: Qemu
-
-   Qemu (Credit: qemu.org)
```
Comment (enjiamai) [inline boards/x86/qemu_x86/doc/index.rst:16]: Hi @gmarull , just double check it. The qemu logo photo will be removed, right?

**Review by enjiamai — DISMISSED** [review]

File: boards/x86/qemu_x86/doc/index.rst (hunk 2 above)
Comment (gmarull) [reply boards/x86/qemu_x86/doc/index.rst:16]: yes, I removed all instances of it (other logos as well)

**Review by povergoing — DISMISSED** [review]
LGTM

**Comment by evgeniy-paltsev** [comment]
Hi @gmarull ,

> The script resizes images to the specified width in rst, and if you check the docs, you'll see it actually looks the same:
So, in this case, it looks like the width in the rst needs to be fixed.

Ok, I'll fix it in the rst files. May I ask you to postpone changes for ARC boards (`boards/arc/*`) till than?

Thanks!

**Comment by gmarull** [comment]
> Hi @gmarull ,
> 
> > The script resizes images to the specified width in rst, and if you check the docs, you'll see it actually looks the same:
> > So, in this case, it looks like the width in the rst needs to be fixed.
> 
> Ok, I'll fix it in the rst files. May I ask you to postpone changes for ARC boards (`boards/arc/*`) till than?
> 
> Thanks!

I have fixed it, please check latest changeset.

**Comment by evgeniy-paltsev** [comment]
> I have fixed it, please check latest changeset.

For mikrobus_header.jpg yes, but I also want to fix it for other ARC boards. I.e: [this](https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/75bdfcc360debdaa40b506ee1285c60d9ae35e15/boards/arc/emsdp/doc/emsdp.jpg) doesn't look really redable.

As previously we could simply open original image size by one mouse click in the documentation that wasn't the issue. But as we reducing original image quality as well - now it is an issue.

I'll prepare PR for ARC boards documentation in a few days.

**Comment by evgeniy-paltsev** [comment]
BTW, looking at the original issue we trying to solve I'm wondering if we are doing the right thing?

After this PR get merged the git repo will contain both images - the original sized ones and the resized, which came from this PR, so it will **increase** overall repo size.

And (from my understanding) it's the thing we won't be able fix after this PR get merged (if we are't talking about repo history rewrite)

@gmarull what do you think?

**Comment by gmarull** [comment]
> BTW, looking at the original issue we trying to solve I'm wondering if we are doing the right thing?
> 
> After this PR get merged the git repo will contain both images - the original sized ones and the resized, which came from this PR, so it will **increase** overall repo size.
> 

Yes, that is actually mentioned in the commit message. Still, there are some advantages like:

- Reduces the repository archive size
- Reduces the size of shallow clones (~130MB down to ~53MB)
- Reduces documentation artifacts size

It is unfortunate that we ended up in such situation because there's no way back as I guess we won't do history re-writes.

> And (from my understanding) it's the thing we won't be able fix after this PR get merged (if we aren't talking about repo history rewrite)
> 
> @gmarull what do you think?

**Comment by soburi** [comment]
LGTM

I've checked 
boards/riscv/longan_nano
boards/arm/arduino_mkrzero

**Review by soburi — DISMISSED** [review]
LGTM

I've checked
boards/riscv/longan_nano
boards/arm/arduino_mkrzero

**Review by enjiamai — DISMISSED** [review]

**Review by povergoing — DISMISSED** [review]

**Review by soburi — DISMISSED** [review]
LGTM

**Review by erwango — DISMISSED** [review]
Didn't parse _all_ STM32 boards, but no issue in the ones I've picked up

**Review by enjiamai — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by cfriedt** [comment]
Oh shoot - I just saw this. 

Actually, some PNG images are smaller than their JPG counterparts. Not a big deal, but I've considered that in the past prior to converting to JPG.

**Pull request merged**